controllers/controllertest/mock_sagemaker_client.go

```diff
@@ -682,23 +675,16 @@ func (m mockSageMakerClient) DescribeTrainingJobRequest(input *sagemaker.Describ
 	mockRequest := m.mockRequestBuilder()
 
 	if nextDescribeTrainingJobResponse.err != nil {
-		mockRequest.Handlers.Send.PushBack(func(r *aws.Request) {
-			r.Error = nextDescribeTrainingJobResponse.err
-		})
-	} else {
-		mockRequest.Handlers.Send.PushBack(func(r *aws.Request) {
-			r.Data = nextDescribeTrainingJobResponse.data
-		})
+		mockRequest.Error = nextDescribeTrainingJobResponse.err
```

Comment:
can you explain this change a bit more?
I thought this is needed so aws client in controller could consume the response. Idea is we push mock responses and when controller pops them per API call instead of making actual API calls

Reply:
main purpose of this block is to return "error" or "data" according to the test case 

Error is return by simply setting the r.Error 
which will be checked during request.Send() by sagemaker api
https://github.com/aws/aws-sdk-go/blob/master/aws/request/request.go#L518

Data is return directly on line 682